protocol/rpcprovider/rpcprovider.go

```diff
@@ -298,7 +301,15 @@ func (rpcp *RPCProvider) SetupProviderEndpoints(rpcProviderEndpoints []*lavasess
 	parallelJobs := len(rpcProviderEndpoints)
 	wg.Add(parallelJobs)
 	disabledEndpoints := make(chan *lavasession.RPCProviderEndpoint, parallelJobs)
+	// validate static spec configuration is used only on a single chain setup.
+	chainIds := make(map[string]struct{})
+	defer func() {
```

Comment:
doing this check in defer is confusing, you can just do len(rpcp.chainMutexes[endpoint.ChainID]) on line 227 (main flow) instead repeatedly checkign whenever you set up an endpoint